commands/new_function.go

```diff
@@ -136,3 +140,49 @@ func validTemplate(lang string) bool {
 
 	return found
 }
+
+func contains(s []string, e string) bool {
+	for _, a := range s {
+		if a == e {
+			return true
+		}
+	}
+	return false
+}
+
+func updateGitignore() (err error) {
```

Comment:
While we may not be able to test the file editing easily, what do you think about extracting a method that does the transform and testing that independently? This would be good for regression.

Reply:
So that extracted functions will only take the file content, updates the content and returns it back ? I should write test for that function ?

Reply:
Sounds like a good approach. That way the function only does 1 thing and can be tested without worrying about editing on disk.

Reply:
Okay .. I will do that. In which file am I supposed to write test and which library to use for testing ?

Reply:
Here's an example - https://github.com/openfaas/faas-cli/blob/master/commands/fetch_templates_test.go

Reply:
Updated and pushed suggested changes.